README.md

```diff
@@ -1,86 +1,149 @@
+
+#### Use the ROCm build tool [rbuild](https://github.com/RadeonOpenCompute/rbuild).
 
-To build dependencies along with MIGraphX
+To install rbuild (sudo may be needed.):
 ```
-rbuild build -d depend --cxx=/opt/rocm/llvm/bin/clang++
+pip3 install https://github.com/RadeonOpenCompute/rbuild/archive/master.tar.gz
 ```
-This builds dependencies in the subdirectory named depend and then builds MIGraphX using these dependencies.
+and pip3 can be installed as `sudo apt update && sudo apt install -y python3-pip`
 
-## Building MIGraphX from source
+Dependencies are listed in files `requirements.txt` and `dev-requirements.txt` in the project diretory. To build the
+dependencies,
+* In ROCm3.3:
 
-## Configuring with cmake
-
-First create a build directory:
+```
+rbuild develop -d depend -B build --cxx=/opt/rocm/bin/hcc
+```
 
+* In ROCm3.7 and later releases:
 
 ```
-mkdir build; 
-cd build;
+rbuild develop -d depend -B build --cxx=/opt/rocm/llvm/clang++
 ```
 
-Next configure cmake. The hcc compiler is required to build the MIOpen backend:
+This builds dependencies in the folder "depend".
 
+Note that if rocBLAS and MIOpen are installed with the command ```sudo apt install -y rocblas miopen-hip``` as mentioned
+above, we can comment out the two lines 
+```
+ROCmSoftwarePlatform/rocBLAS@abd98a2b48b29326ebaef471630786a548622c06
+ROCmSoftwarePlatform/MIOpen@2.4.0
+```
+in the file `requirements.txt` (adding a '#' character at the start of each line).
+Also note that for ROCm3.7 and later release, Ubuntu 18.04 or later releases are needed. Upgrapde to Ubuntu 18.04 can be
+done as:
 
 ```
-CXX=/opt/rocm/llvm/bin/clang++ cmake ..
+sudo apt update
+sudo apt install linux-headers-4.18.0-25-generic linux-image-4.18.0-25-generic  linux-modules-4.18.0-25-generic linux-modules-extra-4.18.0-25-generic -y
 ```
 
-If the dependencies from `install_deps.cmake` was installed to another directory, the `CMAKE_PREFIX_PATH` needs to be set to what `--prefix` was set to from `install_deps.cmake`:
+### Building MIGraphX source
+
+First create a build directory:
 
 
 ```
-CXX=/opt/rocm/llvm/bin/clang++ cmake -DCMAKE_PREFIX_PATH=/some/dir ..
+mkdir build; 
+cd build;
 ```
 
+Next, configure the cmake. If the script [buiild_prereqs.sh](./tools/build_prereqs.sh) is used to install the
+dependencies, then all dependencies are installed at default locations, and MIGraphX can be build as:
+* In ROCm3.3:
 
-#### Changing the cmake configuration
+```
+CXX=/opt/rocm/bin/hcc cmake ..
+```
 
-The configuration can be changed after running cmake by using `ccmake`:
+* In ROCm3.7 or later releases:
 
-` ccmake .. ` **OR** `cmake-gui`: ` cmake-gui ..`
+```
+CXX=/opt/rocm/llvm/bin/clang++ cmake ..
+```
 
-## Building the library
+If the above rbuild command was used to build and install the dependencies, then all dependencies are in the folder
+"depend", and the `CMAKE_PREFIX_PATH` needs to be set to the same folder "depend" (full path are needed here.), and the
+command is:
```

Comment:
`rbuild` already creates a build directory and configures it with cmake(unless you used `rbuild prepare` which the readme doesn't mention).